blacksheep/server/websocket.py

```diff
@@ -50,11 +52,12 @@ def __str__(self):
         return f"The client closed the connection. WebSocket close code: {self.code}."
 
 
-class WebSocket:
+class WebSocket(Request):
     def __init__(
         self, scope: MutableMapping[str, Any], receive: Callable, send: Callable
     ):
         assert scope["type"] == "websocket"
+        super().__init__("GET", get_full_path(scope), scope["headers"])
 
         self._scope = scope
```

Comment:
> To truly achieve full compatibility, we should also bind the request to the ASGI scope (this is used occasionally to get information about the client, the server).

I guess you can just rename the `self._scope` attribute to `self.scope` and it will do the trick.